src/MultiRegionClient.php

```diff
@@ -113,6 +139,52 @@ public function getEndpoint()
     }
 
     /**
+     * @param $commandName
+     * @param array $args
+     *
+     * @return array
+     */
+    public function mapRegions($commandName, array $args = [])
+    {
+        return $this->mapRegionsAsync($commandName, $args)->wait();
+    }
+
+    /**
+     * @param string $commandName
+     * @param array $args
+     *
+     * @return \GuzzleHttp\Promise\PromiseInterface
+     */
+    public function mapRegionsAsync($commandName, array $args = [])
+    {
+        $results = [];
+        $regions = $this->partition
+            ->getRegionsForService($this->args['service']);
+        if (empty($regions)) {
+            return new FulfilledPromise([]);
+        }
+
+        $commandGen = function () use ($regions, $commandName, $args) {
+            foreach ($regions as $region) {
+                yield $region => $this->getClientFromPool($region)
+                    ->getCommand($commandName, $args);
+            }
+        };
+        $resultHandler = static function ($result, $region) use (&$results) {
+            $results[$region] = $result;
+        };
+        $pool = new CommandPool($this, $commandGen(), [
```

Comment:
I wonder if exposing any of the CommandPool options is something we should support? What if someone doesn't want to do these operations concurrently or wants to control the concurrency level?